### PR TITLE
Adding migration export scripts

### PR DESCRIPTION
**Pull request by LuteLuthern**
Adding Migration export scripts.

deploy-amlarc.sh: bash script for deploy amlarc on raw aks cluster.
export-service.sh: bash script for migrating aks/aci service to amlarc/mir.
export-service-util.py: python script helper for export and overwrite parameters.

### CODE REVIEW

File: files/export-service-util.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    print(compute_id)
+
+
+def export(ws: Workspace,
+           compute_type: str = None,
+           timeout_seconds: int = None,
+           show_output: bool = True) -> (str, str):
+    """
+    Export all services under target workspace into template and parameters,
+    valid compute_types are "AKS"/"ACI", default to export for both two types.
```
Comment (frcai) [inline files/export-service-util.py:50]: compute type needs align with line 15

File: files/deploy-amlarc.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+# OPTIONAL C) AKS service has private http endpoint
+#az k8s-extension create --cluster-name $cluster_name --cluster-type managedClusters -n $arcml_extension_name \
+#--extension-type Microsoft.AzureML.Kubernetes --scope cluster --configuration-settings enableInference=True allowInsecureConnections=true \
+#privateEndpointILB=True isAKSMigration=true scoringFe.namespace=default \
+#--subscription $subscription_id -g $resource_group --auto-upgrade-minor-version False
+
+
+extension_install_state=$(az k8s-extension show --name $arcml_extension_name  --cluster-type managedClusters --cluster-name "$cluster_name"   --resource-group "$resource_group" --subscription "$subscription_id" | jq -r '.provisioningState')
+echo "$extension_install_state"
+if [[ $extension_install_state == "Succeeded" ]]
```
Comment (mjaow) [inline files/deploy-amlarc.sh:47]: I remember the k8s-extension cli is sync-create and status is showed in provisioningState after some version. If they have installed an old version, they may hit async-create and status is showed in installState.

So does the script depend on version of k8s-extension cli?

File: files/deploy-amlarc.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+echo 'Waiting for feature register...'
+while [ "$(az feature list --query "[?contains(name, 'Microsoft.ContainerService/AKS-ExtensionManager')].[properties.state]" -o json |jq '.[0][0]')" == 'Registered' ]
+do
+  sleep 5
+done
+az provider register -n Microsoft.ContainerService 1> /dev/null
+
+
+# STEP2 Deploy AmlArc extension
+az k8s-extension create --cluster-name $cluster_name --cluster-type managedClusters -n $arcml_extension_name \
```
Comment (mjaow) [inline files/deploy-amlarc.sh:27]: OPTIONAL A) AKS service has public https endpoint

File: files/deploy-amlarc.sh [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+arcml_extension_name="arcml-extension"
+
+ssl_cname="<YOUR_SSL_CNAME>"
+ssl_cert_pem_file="<YOUR_CERT_PEM_FILE>"
+ssl_key_pem_file="<YOUR_KEY_PEM_FILE>"
+
+# STEP1 Register feature providers
+echo 'Register features...'
+az feature register --namespace Microsoft.ContainerService -n AKS-ExtensionManager --subscription "$subscription_id"> /dev/null
```
Comment (mjaow) [inline files/deploy-amlarc.sh:17]: May the output help debug? I see you ignore it

Comment (LuteLuthern) [reply files/deploy-amlarc.sh]: Print outputs to help debug.

File: files/deploy-amlarc.sh (hunk 3 above)
Comment (LuteLuthern) [reply files/deploy-amlarc.sh:28]: Updated

File: files/deploy-amlarc.sh (hunk 2 above)
Comment (LuteLuthern) [reply files/deploy-amlarc.sh:48]: Will mention this in the doc.

File: files/export-service-util.py (hunk 1 above)
Comment (LuteLuthern) [reply files/export-service-util.py:50]: They have aligned with constants, for readable message only.

File: files/export-service-util.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    base_url = _get_mms_url(ws)
+    mms_endpoint = base_url + '/services'
+    webservices = []
+    headers = {'Content-Type': 'application/json'}
+    headers.update(ws._auth_object.get_authentication_header())
+    params = None
+    if compute_type:
+        if compute_type.upper() not in MIGRATION_WEBSERVICE_TYPES:
+            raise WebserviceException('Invalid compute type "{}". Valid options are "{}"'
+                                      .format(compute_type, ",".join(())))
```
[the remark this answers is not in the document]
Comment (LuteLuthern) [reply files/export-service-util.py]: fixed

File: files/deploy-amlarc.sh [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  sleep 5
+done
+az provider register -n Microsoft.ContainerService 1
+
+
+# STEP2 Deploy AmlArc extension
+# OPTION A) AKS service has public https endpoint
+az k8s-extension create --cluster-name $cluster_name --cluster-type managedClusters -n $arcml_extension_name \
+--extension-type Microsoft.AzureML.Kubernetes --scope cluster --configuration-settings enableInference=True isAKSMigration=true \
+scoringFe.namespace=default sslCname=$ssl_cname --config-protected sslCertPemFile=$ssl_cert_pem_file sslKeyPemFile=$ssl_key_pem_file \
```
[the remark this answers is not in the document]
Comment (LuteLuthern) [reply files/deploy-amlarc.sh]: Fixed

File: docs/migrate-aks-service-to-amlarc.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+### Export and deploy AKS service to AmlArc compute
+The `export-service.sh` will run following steps:
+1. Attach your AmlArc deployed AKS cluster as compute target to Workspace.
+2. Export AKS/ACI services as template and upload to your storage.
+3. Download template and modify parameters.
+4. Deploy to AmlArc compute as online-endpoint and online-deployment service.
+> Attention:
+> - Old services will be kept after a successful export and deployment.
+> - You can set online-endpoint name to the same with old service, but DO NOT deploy to the same cluster, a direct migration is recommended for this scenario.
```
Comment (Zhong-J) [inline docs/migrate-aks-service-to-amlarc.md:65]: Endpoint name needs to be unique within workspace. I think user can't set online-endpoint name to the same with old service?

Comment (LuteLuthern) [reply docs/migrate-aks-service-to-amlarc.md:65]: We removed that limit for migration and export&deploy to ensure the compatibility of scoring url.

File: docs/migrate-aks-service-to-amlarc.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+3. Python package: azureml-core >= 1.37.0.
+4. Cli: azure-cli >= 2.30.0, storage-preview >= 0.7.4, k8s-extension >= 1.0.0.
+5. A shell environment (bash is recommended), make sure the VNet connectivity with your cluster, if any.
+
+## Direct migration
+Make sure you have read the [docs](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-attach-arc-kubernetes?tabs=studio#deploy-azure-machine-learning-extension)
+about AmlArc. Check `migrate-aks-service.sh`, fill in all the parameters, and find the right scenarios under OPTIONS. 
+Then migrate your aks service by run:
+```bash
+./export-service-util
```
Comment (frcai) [inline docs/migrate-aks-service-to-amlarc.md:37]: should be ./migrate-aks-service.sh

**Review by Bozhong68 — APPROVED** [review]

**Pull request merged**